### PR TITLE
Selenium Practice Assignment

### PR DESCRIPTION
**Pull request by VishalKTechnocredits**
Hello Amarja and Shraddha, 
Please review my code.

### CODE REVIEW

**Review by AmarjaBTechnocredits — COMMENTED** [review]
In Runnable class, 
 import org.openqa.selenium.WebDriver; -> Remove the unused import.
 =================================
1]In Alert Handling class ,Please remove unused comments.
2] You can optimize the lines of code , by writing for all the x-path
driver.findElement(By.xpath(prop.getProperty("firstnameXpath"))).sendKeys(firstName);

this as 
driver.findElement(By.xpath(prop.getProperty("firstnameXpath"))).sendKeys(prop.getProperty("firstname"));

**Review by AmarjaBTechnocredits — CHANGES_REQUESTED** [review]
All the methods are different assignments and should run independent of each other in Runnable class if I comment the 
1]ah.verifyAlertDemo(); method the other functions fail.Please make it generic or independent of each other
2]When we run the program we are getting below exception
On Registration Page
Alert text is not correct
Exception in thread "main" org.openqa.selenium.UnhandledAlertException: unexpected alert open: {Alert text : Success!}
  (Session info: chrome=72.0.3626.121)
  (Driver info: chromedriver=2.46.628402 (536cd7adbad73a3783fdc2cab92ab2ba7ec361e1),platform=Windows NT 6.3.9600 x86_64) (WARNING: The server did not provide any stacktrace information)
Command duration or timeout: 0 milliseconds: null

Correct the same.

**Review by AmarjaBTechnocredits — COMMENTED** [review]
Validation is missing for the below error messages

String userNameValidationExpected = "Username length should be greater then 5 characters.";
		String passwordExpectedText = "password is mandatory field.";
		String ReTypePasswordExpectedText = "please reenter password";
		String passwordDontMatchExpectedText = "retype password donot match.";

Please add

**Review by AmarjaBTechnocredits — APPROVED** [review]

File: seleniumByTechnocredits/src/seleniumPractise/vishal/AlertHandling.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+		Thread.sleep(3000);
+		driver.findElement(By.xpath(prop.getProperty("firstnameXpath"))).sendKeys(firstName);
+		driver.findElement(By.xpath("//input[@name='ulname']")).sendKeys(lastName);
+		driver.findElement(By.xpath("//input[@name='cmpname']")).sendKeys(company);
+		// driver.findElement(By.xpath("//div[@name='secondSegment'][1]//button[text()='Submit']")).click();
+		click(driver, "//div[@name='secondSegment'][1]//button[text()='Submit']");
+		Thread.sleep(2000);
+		Alert alert = driver.switchTo().alert();
+
+		String expectedText = firstName + " and " + lastName + " and " + company;
```
Comment (KrishnaKTechnocredits) [inline seleniumByTechnocredits/src/seleniumPractise/vishal/AlertHandling.java:40]: Here you can directly write like  String firstName = prop.getProperty("firstname");
No need to create a extra variable.

**Review by KrishnaKTechnocredits — APPROVED** [review]
From next time onwards please take of formatting and code cleaning, don't comment unexpected code just remove them and write proper comments on every logic.

**Pull request merged**